onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selector_action_transformer.cc

```diff
@@ -46,26 +48,49 @@ void DropQDQNodesRules(SelectorActionRegistry& qdq_selector_action_registry) {
+  // Use a separate selectors & actions for MaxPool and Resize.
+  //
+  // They disallow 16-bit types for MaxPool and Resize:
   // int16 MaxPool is not supported by the ONNX specification.
   // int16 Resize is not supported by the ORT implementation (although allowed by ONNX).
+  //
+  // And cannot eliminate the QDQ for MaxPool if the scale is not positive, as a negative scale will change the ordering
+  // of the elements between quantized & de-quantized values.
   std::unique_ptr<NodeSelector> selector_disallow_16bit = std::make_unique<QDQ::DropQDQNodesSelector>(false);
   qdq_selector_action_registry.RegisterSelectorAndAction(drop_action_no_int16_name,
-                                                         {{"MaxPool", {12}},
-                                                          {"Resize", {}}},
+                                                         {{"Resize", {}}},
                                                          std::move(selector_disallow_16bit),
                                                          std::move(drop_action_no_int16));
 
+  std::unique_ptr<NodeSelector> selector_disallow_16bit_and_nonpositive_scale = std::make_unique<QDQ::DropQDQNodesSelector>(false, true, false);
+  qdq_selector_action_registry.RegisterSelectorAndAction(drop_action_no_int16_nor_nonpositive_scale_name,
+                                                         {{"MaxPool", {12}},
+                                                          {"ReduceMin", {}},
+                                                          {"ReduceMax", {}}},
+                                                         std::move(selector_disallow_16bit_and_nonpositive_scale),
+                                                         std::move(drop_action_no_int16_nor_nonpositive_scale));
+
   std::unique_ptr<NodeSelector> selector = std::make_unique<QDQ::DropQDQNodesSelector>(true);
+  // DepthToSpace and SpaceToDepth not included because there are no integer implementations.
+  // https://github.com/microsoft/onnxruntime/issues/21287
   qdq_selector_action_registry.RegisterSelectorAndAction(drop_action_name,
                                                          {{"Gather", {}},
+                                                          {"GatherElements", {}},
                                                           {"Reshape", {}},
                                                           {"Transpose", {}},
                                                           {"Squeeze", {}},
-                                                          {"Unsqueeze", {}}},
+                                                          {"Unsqueeze", {}},
+                                                          {"Flatten", {}},
+                                                          {"Expand", {}},
+                                                          {"Tile", {}},
+                                                          {"Slice", {}}},
```

Comment:
nit: alphabetically sorted would be slightly nicer

Reply:
Done in https://github.com/microsoft/onnxruntime/pull/21376/commits/848f701408614a6ef99be2d0bbe856bfd671f9be

& thanks for taking a look!